indico/modules/users/util.py

```diff
@@ -379,8 +379,15 @@ def merge_users(source, target, force=False):
     logger.info('Successfully merged %s into %s', source, target)
 
 
-def get_color_for_username(username):
-    return user_colors[crc32(username) % len(user_colors)]
+def get_color_for_user_id(user_id):
+    """Calculate a unique color for a user based on their id.
+
+        :param user_id: the user ID (int), or otherwise a string (external search results)
```

Comment:
Just wondering.. wouldn't it be simpler to just use `crc32(str(user_id))` whichw ould work for all kinds of ids?

Reply:
Yeah, I thought of it, but kind of felt dirty doing it since `user_id` could be anything and result in unexpected behavior. But it's not a bad solution either.

Reply:
Sometimes I wish Python was a strongly-typed language 🙊

Reply:
you could use `def get_color_for_user_id(user_id: t.Union[int, str])` ;)
still not really strongly-typed of course, but sphinx uses this information when autodocumenting a function for example